shared/src/akri/instance.rs

```diff
@@ -186,48 +168,34 @@ pub async fn find_instance(
 /// # }
 /// ```
 pub async fn create_instance(
-    instance_to_create: &Instance,
+    instance_to_create: &InstanceSpec,
     name: &str,
     namespace: &str,
     owner_config_name: &str,
     owner_config_uid: &str,
-    kube_client: &APIClient,
-) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    kube_client: &Client,
+) -> Result<(), anyhow::Error> {
     log::trace!("create_instance enter");
-    let akri_instance_type = RawApi::customResource(API_INSTANCES)
-        .group(API_NAMESPACE)
-        .version(API_VERSION)
-        .within(&namespace);
-
-    let kube_instance = KubeAkriInstance {
-        metadata: ObjectMeta {
-            name: name.to_string(),
-            ownerReferences: vec![OwnerReference {
-                apiVersion: format!("{}/{}", API_NAMESPACE, API_VERSION),
-                kind: "Configuration".to_string(),
-                controller: true,
-                blockOwnerDeletion: true,
-                name: owner_config_name.to_string(),
-                uid: owner_config_uid.to_string(),
-            }],
-            ..Default::default()
-        },
-        spec: instance_to_create.clone(),
-        status: None,
-        types: TypeMeta {
-            apiVersion: Some(format!("{}/{}", API_NAMESPACE, API_VERSION)),
-            kind: Some("Instance".to_string()),
-        },
+    let instances_client: Api<Instance> = Api::namespaced(kube_client.clone(), namespace);
+
+    let mut instance = Instance::new(name, instance_to_create.clone());
+    instance.metadata = ObjectMeta {
+        name: Some(name.to_string()),
+        owner_references: Some(vec![OwnerReference {
+            api_version: format!("{}/{}", API_NAMESPACE, API_VERSION),
+            kind: "Configuration".to_string(),
+            controller: Some(true),
+            block_owner_deletion: Some(true),
```

Comment:
do we want to set block_owner_deletion to true?  i'm not sure there is a reason to track ownership if deletion doesn't happen when akri is being uninstalled.

Reply:
hmm ... nevermind ... i see that you are just matching existing functionality.  i need to refresh on owner refs :)

Reply:
(had it backwards ... this setting isn't blocking the owner from deleting ... heh ... the other way around)